src/Kafka/CreateKafkaInstance/StandardInstanceForm.tsx

```diff
@@ -0,0 +1,300 @@
+  const {
+    form,
+    isNameTaken,
+    isNameInvalid,
+    isNameEmpty,
+    isNameError,
+    isFormEnabled,
+    setName,
+  } = useStandardPlanMachine();
+
+  return (
+    <FieldInstanceName
+      value={form.name || ""}
+      validity={
+        isNameTaken
+          ? "taken"
+          : isNameInvalid
+          ? "invalid"
+          : isNameEmpty && isNameError
+          ? "required"
+          : "valid"
+      }
+      isDisabled={!isFormEnabled}
+      onChange={setName}
+    />
+  );
+};
+
+export const ConnectedFieldCloudProvider: VoidFunctionComponent = () => {
+  const {
+    form,
+    capabilities,
+    isProviderError,
+    isFormEnabled,
+    isBillingSingleMarketplace,
+    setProvider,
+  } = useStandardPlanMachine();
+
+  const providers =
+    isBillingSingleMarketplace &&
+    isBillingSingleMarketplace.marketplace !== "rh"
+      ? capabilities.availableProviders.map((p) => ({
+          ...p,
+          isDisabled: p.id !== isBillingSingleMarketplace.marketplace,
+        }))
+      : capabilities.availableProviders;
+
+  return (
+    <FieldCloudProvider
+      isValid={!isProviderError}
+      providers={providers}
+      value={form.provider}
+      isDisabled={!isFormEnabled}
+      onChange={setProvider}
+    />
+  );
+};
+
+export const ConnectedFieldCloudRegion: VoidFunctionComponent = () => {
+  const {
+    form,
+    selectedProvider,
+    selectedSize,
+    isRegionError,
+    isFormEnabled,
+    capabilities,
+    error,
+    setRegion,
+  } = useStandardPlanMachine();
+
+  return (
+    <FieldCloudRegion
+      validity={
+        isRegionError
+          ? "required"
+          : error === "region-unavailable" ||
+            capabilities.instanceAvailability === "regions-unavailable"
+          ? "region-unavailable"
+          : "valid"
+      }
+      regions={selectedProvider?.regions}
+      value={form.region}
+      isDisabled={!isFormEnabled}
+      isSizeUnavailable={selectedSize?.isDisabled || false}
+      onChange={setRegion}
+    />
+  );
+};
+
+export const ConnectedFieldAZ: VoidFunctionComponent = () => {
+  const { isFormEnabled } = useStandardPlanMachine();
+
+  return (
+    <FieldAZ
+      validity={"valid"}
+      options={"multi"}
+      value={"multi"}
+      isDisabled={!isFormEnabled}
+      onChange={() => false} // AZ is defined by the backend, we just visualize the value here
+    />
+  );
+};
+
+export const ConnectedFieldSize: VoidFunctionComponent<
+  Pick<
+    FieldSizeProps,
+    "onLearnHowToAddStreamingUnits" | "onLearnMoreAboutSizes"
+  >
+> = ({ onLearnHowToAddStreamingUnits, onLearnMoreAboutSizes }) => {
+  const {
+    form,
+    sizes,
+    isSizeOverQuota,
+    isSizeDisabled,
+    isSizeError,
+    isSizeLoadingError,
+    isBillingSelectionRequired,
+    isFormEnabled,
+    isLoadingSizes,
+    isLoading,
+    setSize,
+    remainingQuota,
+  } = useStandardPlanMachine();
+
+  return (
+    <FieldSize
+      value={form.size?.quota || 1}
```

Comment:
Will it fine if it take default value 1 if quota is zero ?

Reply:
It can take `undefined` now, so I removed the default 1. In case of undefined, the slider will render as active but without a knob already visible. This is an "impossible" state tho, as if we don't have a size, we are either showing the spinner or error feedback.